test/e2e/collection/namespace_filtering_test.go

```diff
@@ -0,0 +1,182 @@
+		rootDir        string
+	)
+	appNamespace1 := "application-ns1"
+	appNamespace2 := "application-ns2"
+
+	BeforeEach(func() {
+		if _, err = oc.Literal().From("oc create ns %s", appNamespace1).Run(); err != nil {
+			Fail("failed to create namespace")
+		}
+		if _, err = oc.Literal().From("oc create ns %s", appNamespace2).Run(); err != nil {
+			Fail("failed to create namespace")
+		}
+	})
+	BeforeEach(func() {
+		if err := e2e.DeployLogGeneratorWithNamespace(appNamespace1); err != nil {
+			Fail(fmt.Sprintf("Timed out waiting for the log generator 1 to deploy: %v", err))
+		}
+		if err := e2e.DeployLogGeneratorWithNamespace(appNamespace2); err != nil {
+			Fail(fmt.Sprintf("Timed out waiting for the log generator 2 to deploy: %v", err))
+		}
+		rootDir = filepath.Join(filepath.Dir(filename), "..", "..", "..", "/")
+		if elasticsearch, pipelineSecret, err = e2e.DeployAnElasticsearchCluster(rootDir); err != nil {
+			Fail(fmt.Sprintf("Unable to deploy an elastic instance: %v", err))
+		}
+
+		forwarder := &logging.ClusterLogForwarder{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       logging.ClusterLogForwarderKind,
+				APIVersion: logging.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "instance",
+			},
+			Spec: logging.ClusterLogForwarderSpec{
+				Inputs: []logging.InputSpec{
+					{
+						Name: "application-logs",
+						Application: &logging.Application{
+							Namespaces: []string{appNamespace1},
+						},
+					},
+				},
+				Outputs: []logging.OutputSpec{
+					{
+						Name: elasticsearch.Name,
+						Secret: &logging.OutputSecretSpec{
+							Name: pipelineSecret.ObjectMeta.Name,
+						},
+						Type: logging.OutputTypeElasticsearch,
+						URL:  fmt.Sprintf("https://%s.%s.svc:9200", elasticsearch.Name, elasticsearch.Namespace),
+					},
+				},
+				Pipelines: []logging.PipelineSpec{
+					{
+						Name:       "test-app",
+						OutputRefs: []string{elasticsearch.Name},
+						InputRefs:  []string{"application-logs"},
+					},
+				},
+			},
+		}
+		if err := e2e.CreateClusterLogForwarder(forwarder); err != nil {
+			Fail(fmt.Sprintf("Unable to create an instance of clusterlogforwarder: %v", err))
+		}
+
+	})
+
+	GetNamespaces := func() ([]string, error) {
+		options := metav1.ListOptions{
+			LabelSelector: "component=elasticsearch",
+		}
+		app_index := "app-000001"
+		pods, err := e2e.KubeClient.CoreV1().Pods(constants.OpenshiftNS).List(context.TODO(), options)
+		if err != nil {
+			return nil, errors.New("Getting pods error")
+		}
+		if len(pods.Items) == 0 {
+			return nil, errors.New("No pods found for elasticsearch")
+		}
+		log.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+		query := "--query=" + app_index + "/_search?size=10000"
+		stdout, err := e2e.PodExec(constants.OpenshiftNS, pods.Items[0].Name, "elasticsearch", []string{"es_util", query})
+		if err != nil {
+			return nil, errors.New("Pods exec error")
+		}
+		// Get total hits
+		var jsonMap map[string]interface{}
```

Comment:
All of this code is a duplicate to that which lives in a the functional framework.  Please refactor into a common place if needed (e.g. test/helpers/elasticsearch)